pkg/api/pluginproxy/ds_proxy.go

```diff
@@ -138,6 +143,26 @@ func (proxy *DataSourceProxy) getDirector() func(req *http.Request) {
 			req.Header.Add("Authorization", util.GetBasicAuthHeader(proxy.ds.BasicAuthUser, proxy.ds.BasicAuthPassword))
 		}
 
+		// Parse Custom Header
+		if proxy.ds.SecureJsonData != nil {
+			decryptSdj := proxy.ds.SecureJsonData.Decrypt()
+			index := 1
+			for {
+				headerNameSuffix := fmt.Sprintf("httpHeaderName%d", index)
+				headerValueSuffix := fmt.Sprintf("httpHeaderValue%d", index)
+				if jdh := proxy.ds.JsonData.Get(headerNameSuffix); jdh != nil {
```

Comment:
This will cause an infinite loop since `.Get` will never return `nil`
https://github.com/grafana/grafana/blob/master/pkg/components/simplejson/simplejson.go#L148

Please extract this piece of code into a new function and add tests for this feature.